modules/javafx.graphics/src/main/java/javafx/scene/layout/Background.java

```diff
@@ -314,7 +313,7 @@ private Background(List<BackgroundFill> fills, List<BackgroundImage> images, int
             // The common case is to NOT have percent based radii
             final boolean b = fill.getRadii().hasPercentBasedRadii;
             hasPercentFillRadii |= b;
-            if (fill.fill.isOpaque()) {
+            if (fill.getFill().isOpaque()) {
```

Comment:
it was probably ok to leave some fields as package protected: I don't actually know if the compiler replaces the getFill() function call with direct access to the field.

on the other hand, a public API is always a better choice.

Reply:
Once the code is sufficiently often called, it will be optimized and subsequently inline the call. It then becomes equivalent to a direct field access.